src/Extend/LanguagePack.php

```diff
@@ -75,8 +76,15 @@ private function registerLocale(LocaleManager $locales, Extension $extension, $l
             $locales->addCssFile($locale, $file);
         }
 
+        /** @var ExtensionManager $extensions */
+        $extensions = $container->make(ExtensionManager::class);
+
         foreach (new DirectoryIterator($directory) as $file) {
-            if ($file->isFile() && in_array($file->getExtension(), ['yml', 'yaml'])) {
+            $slug = $file->getBasename(".{$file->getExtension()}");
+
+            if ($file->isFile()
+                && in_array($file->getExtension(), ['yml', 'yaml'])
+                && (strstr($slug, '-') || $extensions->isEnabled($slug))) {
```

Comment:
What is the purpose of `strstr($slug, '-')` here?

Reply:
To transform the package name (`vendor/name`) into an extension ID (`vendor-name`).

E.g. `flarum/auth-github` into `flarum-auth-github`. This is the same extension ID system that is used when putting extensions in JS `flarum.extensions` (they are all IDs).

This extension ID is then used to see if the locale has been loaded (keys that look like `vendor-name.admin`) - this _is_ a breaking change as extensions that do not follow this would not have their locale loaded, but... it's necessary for performance improvements. 

We might want to make a method for it though, instead of simply repeating the same code without explaining what it does well.

Hopefully that made sense 🙂.

Reply:
> Hopefully that made sense slightly_smiling_face.

Sorry, it does not make sense for me at all. :P `strstr()` does not make any replacement, it returns substring after first `-`, which then will be cast to boolean. Essentially it will be interpreted as `true` for every extension (since they must contain at least one `-` for separating vendor and package name), which is both cryptic (`strpos() !== 0` seems to be more common and efficient way for searching if string contains another string) and incorrect (it no longer matters if extension is enabled).

Reply:
Oh, I didn't even look at the context of the code, I apologize. I see what you were asking about now.

Yeah, I guess we should be using `strpos` here instead.

> If you only want to determine if a particular needle occurs within haystack, use the faster and less memory intensive function strpos() instead.

Reply:
I think this code misses the negate. Because this conditional attempts to verify the file name is not `core.yml` or `validation.yml`.

Reply:
Then why you just don't use `in_array() to check this? Using `strstr()` is harder to understand and less reliable (I can name a file as `myextensions.yml` and it also always be loaded).

Reply:
What array would you use in this `in_array` conditional @rob006 ?

Reply:
@luceos `core` and `validation`?

Reply:
Yeah also possible. The reason I decided against that is because we agreed to demand filenames equal to the extension shortened identifier (removing flarum-ext-). This code would allow additional files for core. Not sure what we prefer here.

Reply:
> This code would allow additional files for core.

IMO these files should be explicitly whitelisted (this list could be exposed as public constant). Relying on `-` existence in file name allows language pack to use any name for always loaded translations, which may conflict with new conventions added in future. For example in future Flarum could use `extend.yml` file loaded in specific condition, but such file could be already used by some extensions (because Flarum will always load it, so why not?).

Reply:
strstr($slug, '-')

should likely be replaced with

    strpos($slug, '-') === -1

If we don't find the reason why this was dismissed I would be happy to re-open and test this change.